pkg/kubelet/image_manager_test.go

```diff
@@ -32,14 +32,17 @@ import (
 
 var zero time.Time
 
-func newRealImageManager(t *testing.T, policy ImageGCPolicy) (*realImageManager, *dockertools.FakeDockerClient, *cadvisor.Mock) {
+func newRealImageManager(policy ImageGCPolicy) (*realImageManager, *dockertools.FakeDockerClient, *cadvisor.Mock) {
 	fakeDocker := &dockertools.FakeDockerClient{
 		RemovedImages: util.NewStringSet(),
 	}
 	mockCadvisor := new(cadvisor.Mock)
-	im, err := newImageManager(fakeDocker, mockCadvisor, policy)
```

Comment:
I go back and forth on this. We can have all the objects that do housekeeping expose a Start() and have the Kubelet start all of them, or have them start themselves when they're initialized. They'll be in a good, known state and no implementation details leak to the Kubelet (the fact that there are some background threads for example). The downside is that some initialization is done earlier than would otherwise happen and an assumption that calling NewMainKubelet() means you want to start one. I've found that trade-off to be the correct one, but I am happy to hear otherwise.

Today the way we initialize the Kubelet is interesting, we create it, then we start the housekeepings (GC and BirthCry), then we either run the main loop or have it RunOnce. Arguable, we should have all the periodic things start with Run().